### ISSUE TITLE
Add prime constraint to inverse function

### ISSUE THREAD

**Issue by marsella** [GaloisInc/cryptol-specs#100]
The `mp_mod_inv` method in `Common/utils.cry` is used in several places, but its type signature is not correct; it's only a valid function when the modulus is prime. Cryptol has evolved since that function was first written in 2018, so now I see two options to make it correct:
1. Add a `prime p` constraint to the function type constraint
2. Replace all uses with the `recip` and/or `/.` cryptol-native operators on `Z` types.

I did a quick review of all the modules that use this and it looks like they do all use it correctly (e.g. with a prime modulus). If we choose option 1, there will be some cascading work to add the `prime` constraint up the call chain, but that seems worth the precision to me.

That said, if the built-in operations do what I think they do (compute the multiplicative inverse and do modulus-respecting division), we should use those. We might consider improving documentation on those as well, to be explicit about any additional constraints they may have (e.g. does the modulus have to be prime for those operations? I don't know).

- [ ] Pick an option to make the inverse function correct
- [ ] Propagate it throughout the codebase
- [ ] Update any missing documentation / add useful clarifications where necessary

**Comment by RyanGlScott** [comment]
> does the modulus have to be prime for those operations?

Yes. `recip` and `(/.)` use the `Field` instance for `Z p`, which requires `prime p`.

I am in favor of removing `mp_mod_inv` entirely in favor of `recip`. As far as I can tell, `mp_mod_inv` is only used in this one spot:

https://github.com/GaloisInc/cryptol-specs/blob/dd5b2834ac5604b62f349db6df6b34ba10123b5f/Common/EC/PrimeField/PFEC.cry#L366

And it is straightforward to swap it out with `recip`.
